mne/io/pick.py

```diff
@@ -1104,14 +1104,17 @@ def _picks_str_to_idx(info, picks, exclude, with_ref_meg, return_kind,
     # second: match all to channel names
     #
 
-    bad_name = None
+    bad_names = []
     picks_name = list()
     for pick in picks:
         try:
             picks_name.append(info['ch_names'].index(pick))
         except ValueError:
-            bad_name = pick
-            break
+            bad_names.append(pick)
+
+    if bad_names and bad_names[0] not in ["all", "data"]:
+        logger.warning(f'Channel(s) {bad_names} could not be picked, because '
+                       'they are not present in the info instance.')
```

Comment:
Our internal `warn` function is better about emitting warnings
```suggestion
        warn(f'Channel(s) {bad_names} could not be picked, because '
             'they are not present in the info instance.')
```
Can you change it to this, and use a `with pytest.warns(RuntimeWarning, match='not present in the info):` context manager in your new test in `test_channels.py` to show how/when this line will be hit?

Reply:
@larsoner do you mean logger.warn? it is deprecated, I get "DeprecationWarning: The 'warn' method is deprecated, use 'warning' instead" and fail tests

Reply:
No I mean the one in `mne.utils`

Reply:
@larsoner sure done

Reply:
@larsoner I added myself to doc/changes/names.inc, is there any other place I need to change for tests to work?